### ISSUE TITLE
BASE_HREF / Running on subpath doesn't work

### ISSUE THREAD

**Issue by cheddarwhizzy** [argoproj/argo-workflows#6638]
## Summary 

What happened/what you expected to happen?
I should be able to set `server.baseHref` in the official Helm chart and expect to see the UI at `http://localhost:2746/<baseHref>` (when port-forwarding). Or I should be able to curl the main JS file with the BASE_HREF `curl localhost:2746/<baseHref>/main.21185fc80cd4e35bd034.js`

Before we start, around 2/3 of issues can be fixed by one of the following:

* Have you double-checked your configuration? Maybe 30% of issues are wrong configuration.
Yes, nothing seems out of the ordinary.
* Have you tested to see if it is fixed in the latest version? Maybe 20% of issues are fixed by this.
Latest release and latest RC releases both don't work
* Have you tried using the PNS executer instead of Docker? Maybe 50% of artifact related issues are fixed by this.
No, not sure what this means.

## Diagnostics

👀 Yes! We need all of your diagnostics, please make sure you add it all, otherwise we'll go around in circles asking you for it:

What Kubernetes provider are you using? 
Bare Metal (Kubespray)

What version of Argo Workflows are you running? 
3.0.10, 3.1.8, and others
What executor are you running? Docker/K8SAPI/Kubelet/PNS/Emissary
containerRuntimeExecutor=docker
Did this work in a previous version? I.e. is it a regression?
Not sure, just starting to use workflows
Are you pasting thousands of log lines? That's too much information. 

ArgoCD uses the `--rootpath=<base-href>` instead of leveraging the base href html tag and works fine. Port forwarding the pod directly also doesn't allow me to fetch the `main.<hash>.js` using curl. 

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by cheddarwhizzy** [comment]
Here's some additional info

describe server deployment details
```
  Containers:
   argo-server:
    Image:      quay.io/argoproj/argocli:v3.0.10
    Port:       2746/TCP
    Host Port:  0/TCP
    Args:
      server
      --configmap=argocd-workflows-argo-workflows-workflow-controller-configmap
      --loglevel=debug
      --secure=false
    Requests:
      cpu:      100m
      memory:   256Mi
    Readiness:  http-get http://:2746/ delay=10s timeout=1s period=20s #success=1 #failure=3
    Environment:
      IN_CLUSTER:      true
      ARGO_NAMESPACE:   (v1:metadata.namespace)
      BASE_HREF:       /workflows/
    Mounts:
      /tmp from tmp (rw)
```

Configmap
```
data:
  config: |
    containerRuntimeExecutor: docker
    persistence:
      archive: false
      connectionPool:
        maxIdleConns: 100
        maxOpenConns: 0
      nodeStatusOffLoad: false
      postgresql:
        database: argocd
        host: argocd-postgres-postgresql
        passwordSecret:
          key: password
          name: argo-postgres-config
        port: 5432
        tableName: argo_workflows
        userNameSecret:
          key: username
          name: argo-postgres-config
```

Curl'ing the home page.
```
➜ curl localhost:2746/workflows/
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <title>Argo</title>
    <base href="/workflows/">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="robots" content="noindex">
    <link rel="icon" type="image/png" href="assets/favicon/favicon-32x32.png" sizes="32x32">
    <link rel="icon" type="image/png" href="assets/favicon/favicon-16x16.png" sizes="16x16">
</head>

<body>
    <div id="app"></div>
<script type="text/javascript" src="main.21185fc80cd4e35bd034.js"></script></body>

</html>
```

Trying to curl the main JS file:
```
➜ curl localhost:2746/workflows/main.21185fc80cd4e35bd034.js
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <title>Argo</title>
    <base href="/workflows/">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="robots" content="noindex">
    <link rel="icon" type="image/png" href="assets/favicon/favicon-32x32.png" sizes="32x32">
    <link rel="icon" type="image/png" href="assets/favicon/favicon-16x16.png" sizes="16x16">
</head>

<body>
    <div id="app"></div>
<script type="text/javascript" src="main.21185fc80cd4e35bd034.js"></script></body>

</html>
```

**Comment by alexec** [comment]
Have you tried raising this issue with the argo-helm repository?

**Comment by cheddarwhizzy** [comment]
> Have you tried raising this issue with the argo-helm repository?

This isn't a helm chart issue. The chart works just fine. The `server.baseHref` just sets the `BASE_HREF` env var, which sets the `<base href="/some-subpath/">` tag in the html (see above). But the javascript doesn't load. 

If I don't set the BASE_HREF env var, I can load the UI no issues. 

I can get the main.<hash>.js to load if I use ingress with config
```
  rules:
  - host: workflows.mydomain.com
    http:
      paths:
      - backend:
          serviceName: argocd-workflows-argo-workflows-server
          servicePort: 2746
        path: /workflows/?(.*)
        pathType: Prefix
```
and annotation `nginx.ingress.kubernetes.io/rewrite-target: /$1`

But then I can't click around in the UI. It only loads initially. If you're in the argo-workflows slack, I [posted the issue](https://cloud-native.slack.com/archives/C01QW9QSSSK/p1630031785052400) there as well. 

So, to recap: setting `BASE_HREF` env var and port-forwarding the server pod doesn't load the main.js file at all. If I put ingress in front of this same configuration, I'm able to curl the main.js and load the UI initially, but workflows cannot be viewed. There is a strange caveat which I cover in the video I posted to Slack, which is basically the workflow graph, logs, etc don't load unless I bounce the server pod (I know this makes barely any logical sense, but it's true)

**Comment by cheddarwhizzy** [comment]
So I had `proxy-buffering: "on"` in my Ingress Nginx configMap which was causing the problem. I removed that and all good now
